src/core/lib/promise/pipe.h

```diff
@@ -30,9 +33,47 @@
 
 namespace grpc_core {
 
+namespace pipe_detail {
+template <typename T>
+class Center;
+}
+
 template <typename T>
 struct Pipe;
 
+template <typename T>
+class NextResult final {
```

Comment:
Please document what this class represents and how it is used.

Reply:
done